infra/Pos/Reporting/Methods.hs

```diff
@@ -144,20 +144,15 @@ retrieveLogFiles lconfig = fromLogTree $ lconfig ^. lcTree
             iterNext (part, node) = map (first (part :)) $ fromLogTree node
         in curElems ++ concatMap iterNext (lt ^. ltSubloggers . to HM.toList)
 
--- | Pass a list of absolute paths to log files. This function will
--- archive and compress these files and put resulting file into log
--- directory (returning filepath is absolute).
-compressLogs :: (MonadIO m) => [FilePath] -> m FilePath
+-- | Given a list of files (absolute paths), produces in-memory tarxz
+-- archive.
+compressLogs :: (MonadIO m) => [FilePath] -> m ByteString
 compressLogs files = liftIO $ do
     tar <- tarPackIndependently files
     tarxz <-
         BS.concat <$>
         runConduitRes (yield tar .| Lzma.compress (Just 0) .| consume)
-    aName <- getArchiveName
-    withFile aName WriteMode $ \handle -> do
-        BS.hPut handle tarxz
-        hFlush handle
-    pure aName
+    pure tarxz
```

Comment:
```
    a <- b
    pure a
```
is the same as
```
    b
```

Reply:
I wanted to be explicit about what do I produce. It doesn't do any harm.